File: app/src/main/java/com/wlmtest/forecasttestjava/repository/ApiDataParser.java
package com.wlmtest.forecasttestjava.repository;

import com.wlmtest.forecasttestjava.model.api.currentweatherdata.CurrentWeatherData;
import com.wlmtest.forecasttestjava.model.api.currentweatherdata.Weather;
import com.wlmtest.forecasttestjava.model.api.fivedaysforecastdata.City;
import com.wlmtest.forecasttestjava.model.api.fivedaysforecastdata.FiveDaysForecastData;
import com.wlmtest.forecasttestjava.model.pojo.CityForecastPojo;
import com.wlmtest.forecasttestjava.model.pojo.forecastfivedays.CityForecastFiveDaysPojo;
import com.wlmtest.forecasttestjava.model.pojo.forecastfivedays.FiveDaysForecastPojo;
import com.wlmtest.forecasttestjava.utils.Utils;

import java.util.ArrayList;
import java.util.Date;

public class ApiDataParser {

    public static FiveDaysForecastPojo parseFiveDaysForecastDataToFiveDaysForecastPojo(
            FiveDaysForecastData fiveDaysForecastData) {

        if (fiveDaysForecastData == null || fiveDaysForecastData.getList() ==null){
            return null;
        }


        FiveDaysForecastPojo fiveDaysForecastPojo = new FiveDaysForecastPojo();

        java.util.List<com.wlmtest.forecasttestjava.model.api.fivedaysforecastdata.List> lists = fiveDaysForecastData.getList();
        City city = fiveDaysForecastData.getCity();

        for (com.wlmtest.forecasttestjava.model.api.fivedaysforecastdata.List listReference : lists) {
            CityForecastFiveDaysPojo cityForecastFiveDaysPojo = new CityForecastFiveDaysPojo();

            //city name
            cityForecastFiveDaysPojo.setCityName(city.getName());

            //day and month
            cityForecastFiveDaysPojo.setDayMonthReference(Utils.extractDateInfo(listReference.getDtTxt(), false));

            //hour
            cityForecastFiveDaysPojo.setHourReference(Utils.extractDateInfo(listReference.getDtTxt(), true));

            //Max Temp
            cityForecastFiveDaysPojo.setMaxTemp(Utils.removeInfoAfterPoint(Double.toString(listReference.getMain().getTempMax())));

            //Min Temp
            cityForecastFiveDaysPojo.setMinTemp(Utils.removeInfoAfterPoint(Double.toString(listReference.getMain().getTempMin())));


            //Temperature
            cityForecastFiveDaysPojo.setTemperature(Utils.removeInfoAfterPoint(Double.toString(listReference.getMain().getTemp())));

            //Wind Direction
            cityForecastFiveDaysPojo.setWindDirection(listReference.getWind().getDeg());

            //Wind Speed
            cityForecastFiveDaysPojo.setWindSpeed(Double.toString(listReference.getWind().getSpeed()));

            //Icon
            com.wlmtest.forecasttestjava.model.api.fivedaysforecastdata.Weather weather = listReference.getWeather().size() > 0 ? listReference.getWeather().get(0) : null;
            String icon = weather != null ? weather.getIcon() : "";
            cityForecastFiveDaysPojo.setIcon(icon);

            //Message
            String message = weather != null ? weather.getDescription() : "";
            cityForecastFiveDaysPojo.setMessage(Utils.capitalizeFirstWord(message));

            //latitude
            cityForecastFiveDaysPojo.setLat(Double.toString(city.getCoord().getLat()));

            //longitude
            cityForecastFiveDaysPojo.setLon(Double.toString(city.getCoord().getLon()));

            fiveDaysForecastPojo.addCityForecastFiveDaysPojo(cityForecastFiveDaysPojo);
        }

        return fiveDaysForecastPojo;
    }

    public static java.util.List<CityForecastPojo> parseCurrentWeatherDataToCityForecastPojoList(CurrentWeatherData currentWeatherData) {

        java.util.List<CityForecastPojo> cityForecastPojoList = new ArrayList<>();

        java.util.List<com.wlmtest.forecasttestjava.model.api.currentweatherdata.List> lists = currentWeatherData.getList();

        for (com.wlmtest.forecasttestjava.model.api.currentweatherdata.List listReference : lists) {
            CityForecastPojo cityForecastPojo = new CityForecastPojo();

            //city id
            cityForecastPojo.setCityId(String.valueOf(listReference.getId()));

            //city name
            cityForecastPojo.setCityName(listReference.getName());

            //country
            cityForecastPojo.setCountry(listReference.getSys().getCountry());

            //date
            String strDate = Utils.formatDate(new Date(), null);
            cityForecastPojo.setDateReference(strDate);

            //Max Temp
            cityForecastPojo.setMaxTemp(Utils.removeInfoAfterPoint(Double.toString(listReference.getMain().getTempMax())));

            //Min Temp
            cityForecastPojo.setMinTemp(Utils.removeInfoAfterPoint(Double.toString(listReference.getMain().getTempMin())));


            //Temperature
            cityForecastPojo.setTemperature(Utils.removeInfoAfterPoint(Double.toString(listReference.getMain().getTemp())));

            //Wind Direction
            cityForecastPojo.setWindDirection(listReference.getWind().getDeg());

            //Wind Speed
            cityForecastPojo.setWindSpeed(Double.toString(listReference.getWind().getSpeed()));

            //Icon
            Weather weather = listReference.getWeather().size() > 0 ? listReference.getWeather().get(0) : null;
            String icon = weather != null ? weather.getIcon() : "";
            cityForecastPojo.setIcon(icon);

            //Message
            String message = weather != null ? weather.getDescription() : "";
            cityForecastPojo.setMessage(Utils.capitalizeFirstWord(message));

            //latitude
            cityForecastPojo.setLat(Double.toString(listReference.getCoord().getLat()));

            //longitude
            cityForecastPojo.setLon(Double.toString(listReference.getCoord().getLon()));

            cityForecastPojoList.add(cityForecastPojo);
        }

        return cityForecastPojoList;

    }
}

File: app/src/main/java/com/wlmtest/forecasttestjava/repository/ApiDataParser.kt
package com.wlmtest.forecasttestjava.repository

import com.wlmtest.forecasttestjava.model.api.currentweatherdata.CurrentWeatherData
import com.wlmtest.forecasttestjava.model.api.currentweatherdata.Weather
import com.wlmtest.forecasttestjava.model.api.fivedaysforecastdata.City
import com.wlmtest.forecasttestjava.model.api.fivedaysforecastdata.FiveDaysForecastData
import com.wlmtest.forecasttestjava.model.pojo.CityForecastPojo
import com.wlmtest.forecasttestjava.model.pojo.forecastfivedays.CityForecastFiveDaysPojo
import com.wlmtest.forecasttestjava.model.pojo.forecastfivedays.FiveDaysForecastPojo
import com.wlmtest.forecasttestjava.utils.Utils

import java.util.ArrayList
import java.util.Date

object ApiDataParser {
    @JvmStatic
    fun parseFiveDaysForecastDataToFiveDaysForecastPojo(
        fiveDaysForecastData: FiveDaysForecastData?
    ): FiveDaysForecastPojo? {

        if (fiveDaysForecastData == null || fiveDaysForecastData.list == null) {
            return null
        }


        val fiveDaysForecastPojo = FiveDaysForecastPojo()

        val lists = fiveDaysForecastData.list
        val city = fiveDaysForecastData.city

        for (listReference in lists!!) {
            val cityForecastFiveDaysPojo = CityForecastFiveDaysPojo()

            //city name
            cityForecastFiveDaysPojo.cityName = city!!.name

            //day and month
            cityForecastFiveDaysPojo.dayMonthReference = Utils.extractDateInfo(listReference.dtTxt, false)

            //hour
            cityForecastFiveDaysPojo.hourReference = Utils.extractDateInfo(listReference.dtTxt, true)

            //Max Temp
            cityForecastFiveDaysPojo.maxTemp =
                Utils.removeInfoAfterPoint(listReference.main.tempMax)

            //Min Temp
            cityForecastFiveDaysPojo.minTemp =
                Utils.removeInfoAfterPoint(listReference.main.tempMin)


            //Temperature
            cityForecastFiveDaysPojo.temperature =
                Utils.removeInfoAfterPoint(listReference.main.temp)

            //Wind Direction
            cityForecastFiveDaysPojo.setWindDirection(listReference.wind!!.deg!!)

            //Wind Speed
            cityForecastFiveDaysPojo.windSpeed = java.lang.Double.toString(listReference.wind!!.speed!!)

            //Icon
            val weather = if (listReference.weather!!.size > 0) listReference.weather!!.get(0) else null
            val icon = if (weather != null) weather.icon else ""
            cityForecastFiveDaysPojo.icon = icon

            //Message
            val message = if (weather != null) weather.description else ""
            cityForecastFiveDaysPojo.message = Utils.capitalizeFirstWord(message!!)

            //latitude
            cityForecastFiveDaysPojo.lat = java.lang.Double.toString(city.coord!!.lat!!)

            //longitude
            cityForecastFiveDaysPojo.lon = java.lang.Double.toString(city.coord!!.lon!!)

            fiveDaysForecastPojo.addCityForecastFiveDaysPojo(cityForecastFiveDaysPojo)
        }

        return fiveDaysForecastPojo
    }

    @JvmStatic
    fun parseCurrentWeatherDataToCityForecastPojoList(currentWeatherData: CurrentWeatherData): List<CityForecastPojo> {

        val cityForecastPojoList = ArrayList<CityForecastPojo>()

        val lists = currentWeatherData.list

        for (listReference in lists!!) {
            val cityForecastPojo = CityForecastPojo()

            //city id
            cityForecastPojo.cityId = listReference.id.toString()

            //city name
            cityForecastPojo.cityName = listReference.name

            //country
            cityForecastPojo.country = listReference.sys!!.country

            //date
            val strDate = Utils.formatDate(Date(), null)
            cityForecastPojo.setDateReference(strDate)

            //Max Temp
            cityForecastPojo.maxTemp =
                Utils.removeInfoAfterPoint(listReference.main.temp_max)

            //Min Temp
            cityForecastPojo.minTemp =
                Utils.removeInfoAfterPoint(listReference.main.temp_min)


            //Temperature
            cityForecastPojo.temperature =
                Utils.removeInfoAfterPoint(listReference.main.temp)

            //Wind Direction
            cityForecastPojo.setWindDirection(listReference.wind!!.deg!!)

            //Wind Speed
            cityForecastPojo.windSpeed = java.lang.Double.toString(listReference.wind!!.speed!!)

            //Icon
            val weather = if (listReference.weather!!.size > 0) listReference.weather!!.get(0) else null
            val icon = if (weather != null) weather.icon else ""
            cityForecastPojo.icon = icon

            //Message
            val message = if (weather != null) weather.description else ""
            cityForecastPojo.message = Utils.capitalizeFirstWord(message!!)

            //latitude
            cityForecastPojo.lat = java.lang.Double.toString(listReference.coord!!.lat!!)

            //longitude
            cityForecastPojo.lon = java.lang.Double.toString(listReference.coord!!.lon!!)

            cityForecastPojoList.add(cityForecastPojo)
        }

        return cityForecastPojoList

    }
}
